imagecraft/pack/image.py

```diff
@@ -98,16 +99,19 @@ def attach_loopdev(self) -> str:
             emit.debug(
                 f"Attached image {self.disk_path} as loop device {self.loop_device}"
             )
-        return self.loop_device
+        try:
+            yield self.loop_device
+        finally:
+            self._detach_loopdevs()
```

Comment:
Why is the context manager function `attach_loopdev()` (singular) but it ends with `_detach_loopdevs()` (plural)? I attach a device and then remove all attached devices? Shouldn't we detach only the one that was attached?

Reply:
We do detach **only** loop devices associated to the given "back-file". Even though we only create a single loop device and attach the file to it, there is no way to guaranty the file has not been attached a second time by another process. So with the current approach listing loop devices associated to the file could yield multiple results. The current approach is a rather safe one because it makes sure to detach everything to avoid corrupting the file. 
Another approach would be to use `self.loop_device` to only detach this specific loop device.